frontend/public/components/cluster-settings/cluster-settings.tsx

```diff
@@ -69,6 +88,54 @@ const cancelUpdate = (cv: ClusterVersionKind) => {
 
 export const clusterAutoscalerReference = referenceForModel(ClusterAutoscalerModel);
 
+const getMachineConfigPoolByName = (
+  machineConfigPools: MachineConfigPoolKind[],
+  name: string,
+): MachineConfigPoolKind => {
+  return machineConfigPools?.find((mcp) => mcp.metadata.name === name);
+};
+
+const getStartedTimeForCVDesiredVersion = (
+  cv: ClusterVersionKind,
+  desiredVersion: string,
+): string => {
+  const desiredHistory: UpdateHistory = cv?.status?.history?.find(
+    (update) => update.version === desiredVersion,
+  );
+  return desiredHistory?.startedTime;
+};
+
+const getUpdatingTimeForMCP = (machineConfigPool: MachineConfigPoolKind): string => {
+  const updatingCondition = machineConfigPool?.status?.conditions.find(
+    (condition) => condition.type === MachineConfigPoolConditionType.Updating,
+  );
+  return updatingCondition?.lastTransitionTime;
+};
+
+const getUpdatedOperatorCount = (
+  clusterOperators: ClusterOperator[],
+  desiredVersion: string,
+): number => {
+  return clusterOperators?.filter((operator) => {
+    const progressingCondition = operator?.status?.conditions.find(
+      (condition) => condition.type === 'Progressing',
+    );
+    return (
+      getClusterOperatorVersion(operator) === desiredVersion &&
+      progressingCondition?.status === 'False'
+    );
+  })?.length;
+};
+
+const getIsUpdating = (clusterOperators: ClusterOperator[], cv: ClusterVersionKind): boolean => {
+  const desiredVersion = getDesiredClusterVersion(cv);
+  const updatedOperatorsCount = getUpdatedOperatorCount(clusterOperators, desiredVersion);
+  return updatedOperatorsCount > 0;
```

Comment:
The cluster-version operator looks at more than just ClusterOperator when judging overall update completion.  It's better to look at [ClusterVersion's `Progressing` status][1], as `oc` does [here][2].

[1]: https://github.com/openshift/api/blob/98a218722c5ef3e3d8c612a04968666a63e1e3fe/config/v1/types_cluster_version.go#L115
[2]: https://github.com/openshift/oc/blob/a38a015e8e40a4923e87864177683e60d01e8b11/pkg/cli/admin/upgrade/upgrade.go#L437-L438

Reply:
We're sort of short circuiting the ClusterVersion here in order to cover the gap between when the update starts and the first operator is updated.  It's a bit of sleight of hand so the user doesn't stare at an empty checklist for a long time and wonder if anything is happening.

Reply:
How does that handle the tail of the update, though?

Reply:
Hey @wking, we've gotten customer feedback that cluster admins have to look at several resources like cluster operators and nodes to understand what's happening during an upgrade. This is a step to bring some of that together in one place so you can see operator, master, and worker progress separately. I realize that CVO looks at more and the number won't track precisely with the overall percentage complete from the Progressing condition, but it gives some insight into what's happening when during an upgrade, more than can be conveyed with a single number.

In particular, we've heard that worker node progress is important to customers since it can impact workloads running on the cluster (even though workers don't block the overall upgrade from completing).

If the CVO is able to give us a more precise breakdown, we'd love to be able to show that here.

Reply:
> How does that handle the tail of the update, though?

Not sure I entirely understand the question.  We only do this when the checklist is visible.  When the CVO status changes to `UpToDate`, we show that under Cluster Status.

Reply:
After revisiting this, @spadgett, @megan-hall, and I decided a single message of `Update to {version} in progress` is the better option.